packages/console/src/components/AuditLogTable/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { LogDTO, LogResult } from '@logto/schemas';
+import { conditionalString } from '@silverhand/essentials';
+import classNames from 'classnames';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import useSWR from 'swr';
+
+import ApplicationName from '@/components/ApplicationName';
+import Pagination from '@/components/Pagination';
+import TableEmpty from '@/components/Table/TableEmpty';
+import TableError from '@/components/Table/TableError';
+import TableLoading from '@/components/Table/TableLoading';
+import UserName from '@/components/UserName';
+import { RequestError } from '@/hooks/use-api';
+import * as tableStyles from '@/scss/table.module.scss';
+
+import ApplicationSelector from './components/ApplicationSelector';
+import EventName from './components/EventName';
+import EventSelector from './components/EventSelector';
+import * as styles from './index.module.scss';
+
+const pageSize = 20;
+
+type Props = {
+  userId?: string;
+};
+
+const AuditLogTable = ({ userId }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [query, setQuery] = useSearchParams();
+  const pageIndex = Number(query.get('page') ?? '1');
+  const event = query.get('event');
+  const applicationId = query.get('applicationId');
+  const { data, error, mutate } = useSWR<[LogDTO[], number], RequestError>(
+    `/api/logs?page=${pageIndex}&page_size=${pageSize}${conditionalString(
+      event && `&logType=${event}`
+    )}${conditionalString(applicationId && `&applicationId=${applicationId}`)}${conditionalString(
+      userId && `&userId=${userId}`
+    )}`
+  );
+  const isLoading = !data && !error;
+  const navigate = useNavigate();
+  const [logs, totalCount] = data ?? [];
+
+  return (
+    <>
+      <div className={styles.filter}>
+        <div className={styles.title}>{t('logs.filter_by')}</div>
+        <div className={styles.eventSelector}>
+          <EventSelector
+            value={event ?? undefined}
+            onChange={(value) => {
+              setQuery({ event: value ?? '' });
+            }}
+          />
+        </div>
+        <div className={styles.applicationSelector}>
+          <ApplicationSelector
+            value={applicationId ?? undefined}
+            onChange={(value) => {
+              setQuery({ applicationId: value ?? '' });
+            }}
+          />
+        </div>
+      </div>
+      <div className={classNames(styles.table, tableStyles.scrollable)}>
+        <table className={classNames(!data && tableStyles.empty)}>
```

Comment:
Hmmm, if there's no log records in the table, would the `data` be undefined or empty array?

Reply:
updated